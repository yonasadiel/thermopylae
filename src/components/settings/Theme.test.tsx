import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import ThemeSettings from './Theme';

describe('ThemeSettings', () => {
    test('should render correctly', () => {
        const result = render(<ThemeSettings />);
        expect(result.baseElement).toMatchSnapshot();
    });

    test('should render correctly when particle is not enabled', async () => {
        const result = render(<ThemeSettings />);
        const elem = await result.findByLabelText('Particles')
        elem.click();
        expect(result.baseElement).toMatchSnapshot();
    });
});