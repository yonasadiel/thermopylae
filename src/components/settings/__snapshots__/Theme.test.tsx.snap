// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSettings should render correctly 1`] = `
<body>
  <div>
    <div
      class="theme-settings"
    >
      <div
        class="input-group"
      >
        <label>
          Style
        </label>
        <p>
          <input
            type="color"
            value="black"
          />
          <span>
            Background Color
          </span>
        </p>
        <p>
          <input
            type="color"
            value="white"
          />
          <span>
            Color
          </span>
        </p>
      </div>
      <div
        class="input-group"
      >
        <label>
          Background Image
        </label>
        <input
          type="file"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="particles-enabled"
        >
          Particles
        </label>
        <input
          checked=""
          id="particles-enabled"
          type="checkbox"
        />
        <span>
          Enabled
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`ThemeSettings should render correctly when particle is not enabled 1`] = `
<body>
  <div>
    <div
      class="theme-settings"
    >
      <div
        class="input-group"
      >
        <label>
          Style
        </label>
        <p>
          <input
            type="color"
            value="black"
          />
          <span>
            Background Color
          </span>
        </p>
        <p>
          <input
            type="color"
            value="white"
          />
          <span>
            Color
          </span>
        </p>
      </div>
      <div
        class="input-group"
      >
        <label>
          Background Image
        </label>
        <input
          type="file"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="particles-enabled"
        >
          Particles
        </label>
        <input
          checked=""
          id="particles-enabled"
          type="checkbox"
        />
        <span>
          Enabled
        </span>
      </div>
    </div>
  </div>
</body>
`;
